apps/hash-graph/lib/graph/src/knowledge/entity/mod.rs

```diff
@@ -173,8 +173,10 @@ impl LinkData {
 // TODO: deny_unknown_fields on other structs
 #[serde(deny_unknown_fields, rename_all = "camelCase")]
 pub struct EntityMetadata {
-    edition_id: EntityEditionId,
-    version: EntityVersion,
+    #[serde(rename = "editionId")]
+    record_id: EntityRecordId,
+    #[serde(rename = "version")]
+    revision: EntityRevisionVersion,
     #[schema(value_type = String)]
     entity_type_id: VersionedUri,
     #[serde(rename = "provenance")]
```

Comment:
The PR that will be opened soon on the BP will have the following shape and names for this:

```ts
export type EntityMetadata<Temporal extends boolean> = {
  recordId: EntityRecordId;
  entityTypeId: VersionedUri;
  temporalVersioning: Temporal extends true
    ? EntityTemporalVersioningMetadata
    : never;
};
```

(You can ignore the conditional bit as in HASH it will always be versioned)

I'd probably suggest we revert the `revision` metadata field rename in this PR, and address it in a follow-up once we've decided on the terminology in the BP. I feel pretty strongly that we _don't_ call the field `revision` here, because a revision of the entity _is a specific instance_ of it, it's not the identifier of that instance. 

I'm suggesting we revert rather than modifying, because I really don't like the term `EntityTemporalVersioningMetadata` and I'd like to see us find something better, but that discussion should really happen in the upcoming PR and I appreciate we don't want to block this PR on that as there are more things stacked on top of it

Reply:
This was also a thing I was unsure about and the main reason why I opened #1999 to only change the required things to move forward.